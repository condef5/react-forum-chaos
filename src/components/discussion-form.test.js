import React from "react";
import { render, fireEvent, waitForElement } from "@testing-library/react";
import DiscussionForm from "./discussion-form";

test("DiscussionForm component", async () => {
  Date.now = jest.fn(() => new Date("2019-06-13T22:50:18.315Z"));
  const handleCreateDisccusion = jest.fn();

  const { getByLabelText, asFragment } = render(
    <DiscussionForm
      handleCreateDisccusion={handleCreateDisccusion}
      handleCloseModal={jest.fn()}
    />
  );

  expect(asFragment()).toMatchSnapshot();

  const titleInput = await waitForElement(() =>
    getByLabelText("Write the title of the discussion")
  );

  fireEvent.change(titleInput, { target: { value: "React is great" } });

  const bodyInput = getByLabelText("Write the body of the discussion");
  fireEvent.change(bodyInput, { target: { value: "Space in the morning" } });

  fireEvent.submit(titleInput);

  expect(handleCreateDisccusion).toHaveBeenCalledWith({
    id: new Date("2019-06-13T22:50:18.315Z").getTime(),
    title: "React is great",
    body: "Space in the morning",
    author: "condef5",
    date: new Date("2019-06-13T22:50:18.315Z").toISOString()
  });
});
