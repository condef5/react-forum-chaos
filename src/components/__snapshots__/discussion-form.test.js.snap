// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home component 1`] = `
<DocumentFragment>
  .emotion-0 {
  font-size: 2rem;
  font-weight: normal;
  margin: 0;
}

.emotion-2 {
  margin: 2em 0;
}

.emotion-1 {
  background: none;
  border: 1px solid #eaeaea;
  border-radius: .25rem;
  box-sizing: border-box;
  display: block;
  font-size: 1rem;
  padding: .5rem;
  color: #34495e;
  width: 100%;
}

.emotion-1:focus {
  outline: none;
  border-color: rgba(255,102,0,0.5);
}

.emotion-3 {
  background: none;
  border: 1px solid #eaeaea;
  border-radius: .25rem;
  box-sizing: border-box;
  display: block;
  font-size: 1rem;
  padding: .5rem;
  color: #34495e;
  width: 100%;
}

.emotion-3:focus {
  outline: none;
  border-color: rgba(255,102,0,0.5);
}

.emotion-5 {
  background-color: #f60;
  border: 1px solid #f60;
  border-radius: .25rem;
  color: white;
  cursor: pointer;
  font-size: .8rem;
  padding: .75rem 0;
  -webkit-transition: all 200ms ease;
  transition: all 200ms ease;
  text-align: center;
  text-transform: uppercase;
  width: 100%;
}

.emotion-5:hover {
  background-color: white;
  color: #f60;
}

<form>
    <h2
      class="emotion-0"
    >
      New Discussion
    </h2>
    <div
      class="emotion-2"
    >
      <input
        aria-label="Write the title of the discussion"
        class="emotion-1"
        name="title"
        placeholder="Discussion title"
        required=""
        type="text"
        value=""
      />
    </div>
    <div
      class="emotion-2"
    >
      <textarea
        aria-label="Write the body of the discussion"
        class="emotion-3"
        name="body"
        placeholder="Body title"
        required=""
        rows="4"
      />
    </div>
    <button
      class="emotion-5"
    >
      Create employee
    </button>
  </form>
</DocumentFragment>
`;
